AzureIndexer/Stratis.Bitcoin.Features.AzureIndexer/Checkpoint.cs

```diff
@@ -65,18 +73,51 @@ public BlockLocator BlockLocator
 
         public bool SaveProgress(ChainedBlock tip)
         {
-            return SaveProgress(tip.GetLocator());
+            this.logger.LogTrace("()");
+
+            bool progress = SaveProgress(tip.GetLocator());
+
+            this.logger.LogTrace("(-):{0}", progress);
+            return progress;
         }
         public bool SaveProgress(BlockLocator locator)
         {
+            this.logger.LogTrace("()");
+
             _BlockLocator = locator;
             try
             {
-                return SaveProgressAsync().Result;
+                Task<bool> savingTask = Task.Run(new Func<Task<bool>>(async () =>
+                {
+                    this.logger.LogTrace("()");
+
+                    var saving = SaveProgressAsync();
+                    var timeout = Task.Delay(50000);
+
+                    await Task.WhenAny(saving, timeout).ConfigureAwait(false);
+
+                    if (saving.IsCompleted)
+                    {
+                        this.logger.LogTrace("Saving completed.");
+                        this.logger.LogTrace("(-):{0}", saving.Result);
+                        return saving.Result;
+                    }
+
+                    this.logger.LogTrace("(-):TIMEOUT");
+                    return false;
+                }));
+                
+                bool result = savingTask.GetAwaiter().GetResult();
+
+                this.logger.LogTrace("(-):{0}", result);
+                return result;
             }
-            catch (AggregateException aex)
+            catch (Exception aex)
```

Comment:
Just ex? Or should it still be an AggregateException?

Reply:
Yeah. After the changes above we would like to log any exception